src/controllers/mainController.ts

```diff
@@ -1411,4 +1418,38 @@ export default class MainController implements vscode.Disposable {
 	public onClearAzureTokenCache(): void {
 		this.connectionManager.onClearTokenCache();
 	}
+
+	private ExecutionPlanCustomEditorProvider = class implements vscode.CustomTextEditorProvider {
+		context: vscode.ExtensionContext;
+		service: any;
+        constructor(
+			context: vscode.ExtensionContext,
+			service: ExecutionPlanService,
+        ) {
+			this.context = context;
+			this.service = service;
+        }
+
+        public async resolveCustomTextEditor(
+            document: vscode.TextDocument
+        ): Promise<void> {
+			await vscode.commands.executeCommand('workbench.action.closeActiveEditor');
```

Comment:
what is this about?  Do we need to manually close the XML file or something?

Reply:
by default, vscode first opens the document in it's text editor, and then calls the custom editor to handle it. when we make a new react webview, it opens in another tab, so I'm closing the old tab and then opening the react view. I'm trying to see if there's a way to open the content in the existing tab

Reply:
I'm a little worried about race conditions with this snippet since we aren't explicitly expressing which editor to close.  For example, if the active editor changes between the time the temp doc is opened and this call is run.  Ideally, we could use the initial document directly if possible, or if not at least use an API that allows us to provide the URI of the document to close.

Reply:
addressed this by moving the closing command to within the onDidOpenTextDocument function so it's guaranteed to close the correct file